### ISSUE TITLE
Comment on a line after closing heredoc causes "unexpected '#'"

### ISSUE THREAD

**Issue by 4x0v7** [hadolint/hadolint#960]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
No error

### Actual behavior
`Dockerfile:6:1 unexpected '#' expecting a new line followed by the next instruction`

### Steps to reproduce the behavior

Create a Dockerfile as shown, with a `# comment` on the line after `EOF`

If the comment is _after_ a newline, the error is not shown.

For example this doesn't trigger the error:
```Dockerfile
# syntax=docker/dockerfile:1
FROM alpine:3.17
RUN <<EOF
    echo "test"
EOF

# comment
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1
FROM alpine:3.17
RUN <<EOF
    echo "test"
EOF
# comment

```

**Additional environment details (OS, stack version, etc.)**

**Comment by jsaraiva** [comment]
Can also confirm that this happens with any character that immediately follows a heredoc. It does not need to be a comment.
For example, this triggers the error:
`
RUN <<EOF
  yarn install --frozen-lockfile
  yarn cache clean
EOF
COPY --from=development__scratchpad /app ./
`

Can also confirm that adding a newline after the EOF avoids the error. The following does not show any error:
`
RUN <<EOF
  yarn install --frozen-lockfile
  yarn cache clean
EOF

COPY --from=development__scratchpad /app ./
`
